lib/repository.py

```diff
@@ -129,15 +129,23 @@ def __init__(self, repo_path, *args, **kwargs):
     def name(self):
         return os.path.basename(self.working_tree_dir)
 
-    def checkout(self, ref_name):
+    def checkout(self, ref_name, ref_specs=None):
         """
         Check out the reference name, resetting the index state.
         The reference may be a branch, tag or commit.
+
+        Args:
+            ref_name (str): name of the reference. May be a branch, tag,
+                commit ID, etc.
+            ref_specs ([str]): pattern mappings from remote references to
+                local references, as specified by Git.
```

Comment:
maybe it's worth it to include https://git-scm.com/book/id/v2/Git-Internals-The-Refspec ?